src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -2,32 +2,83 @@
 
 import android.annotation.SuppressLint;
 import android.content.ComponentName;
+import android.graphics.Bitmap;
+import android.net.http.SslCertificate;
+import android.net.http.SslError;
 import android.os.Bundle;
 import android.os.IBinder;
+import android.util.Log;
 import android.view.View;
+import android.webkit.SslErrorHandler;
 import android.webkit.WebView;
 import android.webkit.WebViewClient;
 import android.widget.ProgressBar;
 
 import com.nutomic.syncthingandroid.R;
+import com.nutomic.syncthingandroid.syncthing.RestApi;
 import com.nutomic.syncthingandroid.syncthing.SyncthingService;
 
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Holds a WebView that shows the web ui of the local syncthing instance.
  */
 public class WebGuiActivity extends SyncthingActivity implements SyncthingService.OnWebGuiAvailableListener {
 
+    private static final String TAG = "WebGuiActivity";
+
     private WebView mWebView;
 
     private View mLoadingView;
 
+    private X509Certificate mCaCert;
+
     /**
-     * Hides the loading screen and shows the WebView once it is fully loaded.
+     * Hides the loading screen and shows the WebView once it is partially loaded.
      */
     private final WebViewClient mWebViewClient = new WebViewClient() {
 
+        /**
+         * Catch (self-signed) SSL errors. Still accept them if they correspond to Syncthing's certificate.
+         * Uses reflection to access the private mX509Certificate field of SslCertificate
+         * (the Android API does not provide a way to properly validate the certificate).
```

Comment:
Javadoc is usually one line, and then a new paragraph with additional info. Makes it easier to read.

https://en.wikipedia.org/wiki/Javadoc#Overview_of_Javadoc